src/server/api/creation.ts

```diff
@@ -34,8 +34,8 @@ export const creationRoutes = (cfg: APIConfig, store: Store, apsystem: ActivityP
     }
 
     const info = request.body
-    await store.forActor(actor).setInfo(info)
     await apsystem.announcements.announce(actor, info)
+    await store.forActor(actor).setInfo(info)
```

Comment:
I worry this brings a race condition where we announce the actor exists before it's fully registered. We should detect if actor is new outside of the `announce` method and set the info before making the announcement.